plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -822,6 +822,26 @@ var _ = Describe("validator", func() {
 				Expect(err).To(BeInternalServerError())
 			})
 
+			It("should reject because for a new Shoot both cloudProfileName and cloudProfile are provided", func() {
+				shoot.Spec.SeedName = nil
+				Expect(coreInformerFactory.Core().V1beta1().Projects().Informer().GetStore().Add(&project)).To(Succeed())
+				Expect(coreInformerFactory.Core().V1beta1().CloudProfiles().Informer().GetStore().Add(&cloudProfile)).To(Succeed())
+				Expect(coreInformerFactory.Core().V1beta1().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+				Expect(coreInformerFactory.Core().V1beta1().SecretBindings().Informer().GetStore().Add(&secretBinding)).To(Succeed())
+				Expect(securityInformerFactory.Security().V1alpha1().CredentialsBindings().Informer().GetStore().Add(&credentialsBinding)).To(Succeed())
+
+				shoot.Spec.CloudProfileName = ptr.To("profile")
+				shoot.Spec.CloudProfile = &core.CloudProfileReference{
+					Kind: "CloudProfile",
+					Name: "profile",
+				}
+				attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, userInfo)
+				err := admissionHandler.Admit(ctx, attrs, nil)
+
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring("either cloudProfileName or cloudProfile reference"))
+			})
+
 			It("should reject because the referenced seed was not found", func() {
```

Comment:
Do you want to add further tests regarding cloud profile changes, at least a rudimentary one?

Reply:
Done, thanks!